code/modules/organs/internal/lungs/lung.dm

```diff
@@ -0,0 +1,120 @@
+
+///////////////////////
+// LUNG ORGAN
+///////////////////////
+
+/datum/organ/internal/lungs
+	name = "lungs"
+	parent_organ = "chest"
+	removed_type = /obj/item/organ/lungs
+
+	// VG now delegates breathing to the appropriate organ.
+
+	// DEFAULTS FOR HUMAN LUNGS:
+	var/list/datum/lung_gas/gasses = list(
+		new /datum/lung_gas/metabolizable("oxygen",            min_pp=16, max_pp=140),
+		new /datum/lung_gas/waste("carbon_dioxide",            max_pp=10),
+		new /datum/lung_gas/toxic("toxins",                    max_pp=0.5, max_pp_mask=5, reagent_id="plasma", reagent_mult=10),
+		new /datum/lung_gas/sleep_agent("/datum/gas/sleeping_agent", trace_gas=1, min_giggle_pp=0.15, min_para_pp=1, min_sleep_pp=5),
+	)
+
+	var/inhale_volume = BREATH_VOLUME
+	var/exhale_moles = 0
+
+/datum/organ/internal/lungs/proc/gasp()
+	owner.emote("gasp")
+
+/datum/organ/internal/lungs/proc/handle_breath(var/datum/gas_mixture/breath, var/mob/living/carbon/human/H)
+	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/datum/species/proc/handle_breath() called tick#: [world.time]")
+
+	// NOW WITH MODULAR GAS HANDLING RATHER THAN A CLUSTERFUCK OF IF-TREES FOR EVERY SNOWFLAKE RACE
+	//testing("Ticking lungs...")
+
+	// First, we consume air.
+	for(var/datum/lung_gas/G in gasses)
+		G.set_context(src,breath,H)
+		G.handle_inhale()
+
+	// Next, we exhale. At the moment, only /datum/lung_gas/waste uses this.
+	for(var/datum/lung_gas/G in gasses)
+		G.set_context(src,breath,H)
+		G.handle_inhale()
```

Comment:
Wouldn't this be handle_exhale, considering the comment says that "Next, we exhale"?